dnf-docker-test/features/steps/shell_commands.py

```diff
@@ -17,25 +18,15 @@ def step_i_have_dnf_shell_session_opened_with_parameters(context, parameters):
 
 @when('I run dnf shell command "{command}"')
 def step_i_run_dnf_shell_command(context, command):
-    context.cmd_result = None
+    context.cmd_result = CommandResult()
     context.assertion.assertNotEqual(None, context.pexpect_session, "dnf shell session must be opened first")
     context.pexpect_session.sendline(command)
     if command.strip() == 'quit' or command.strip() == 'exit':
         context.pexpect_session.expect(pexpect.EOF)
-        context.cmd_result = context.pexpect_session.before
+        context.cmd_result.stdout = context.pexpect_session.before[len(command):].strip()
```

Comment:
I would need to take a loot at it. The problem is that pexpect is adding extra characters around the message (^M). strip() was an easy method how to get rid of them but I probably need to take a look how it works for multi-line output captured by pexpect.